File: backend-java/src/main/java/com/souschef/dto/user/UserPreferencesResponse.java
package com.souschef.dto.user;

import lombok.Data;
import java.util.ArrayList;
import java.util.List;

@Data
public class UserPreferencesResponse {
    private Long userId;
    private UserPreferences preferences = new UserPreferences();
    
    @Data
    public static class UserPreferences {
        private String dietaryType;
        private List<String> cuisineTypes = new ArrayList<>();
        private String spiceLevel;
        private String cookingTimePreference;
    }
}

File: backend-csharp/DTOs/User/UserPreferencesResponse.cs
namespace SousChefBackend.DTOs.User;

public class UserPreferencesResponse
{
    public long UserId { get; set; }
    public UserPreferences Preferences { get; set; } = new();
}

public class UserPreferences
{
    public string? DietaryType { get; set; }
    public List<string> CuisineTypes { get; set; } = new();
    public string? SpiceLevel { get; set; }
    public string? CookingTimePreference { get; set; }
}

